packages/replay/src/replay.ts

```diff
@@ -807,9 +807,17 @@ export class ReplayContainer implements ReplayContainerInterface {
     } catch (err) {
       this._handleException(err);
 
-      // This means we retried 3 times, and all of them failed
+      // This means we retried 3 times and all of them failed,
+      // or we ran into a problem we don't want to retry, like rate limiting.
       // In this case, we want to completely stop the replay - otherwise, we may get inconsistent segments
       this.stop();
+
+      const hub = getCurrentHub();
+      const client = hub.getClient();
```

Comment:
super-l: Any reason for this over
```suggestion
      const client = getCurrentHub().getClient();
```
?